crates/forge/tests/e2e/packages_validation.rs

```diff
@@ -0,0 +1,149 @@
+
+    cairo_packages_paths
+}
+
+#[test]
+fn check_all_packages_formatting() {
+    let cairo_packages_paths = find_all_cairo_packages_paths();
+
+    for path in cairo_packages_paths {
+        println!("Running `scarb fmt --check` in directory: {path}");
+        let output = ScarbCommand::new()
+            .current_dir(path)
+            .arg("fmt")
+            .arg("--check")
+            .command()
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .output()
+            .expect("Failed to run `scarb fmt --check`");
+
+        assert!(output.status.success(), "`scarb fmt --check` failed");
+    }
+}
+
+fn check_and_lint(package_path: Utf8PathBuf) {
+    println!("Running `scarb check` in directory {package_path}");
+    let check_output = ScarbCommand::new()
+        .current_dir(&package_path)
+        .arg("check")
+        .command()
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .output()
+        .expect("Failed to run `scarb check`");
+    assert!(check_output.status.success(), "`scarb check` failed");
+
+    // TODO(#3149)
+    if cfg!(feature = "scarb_since_2_10") {
+        println!("Running `scarb lint` in directory {package_path}");
+        let lint_output = ScarbCommand::new()
+            .current_dir(package_path)
+            .arg("lint")
+            .command()
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .output()
+            .expect("Failed to run `scarb lint`");
+        assert!(lint_output.status.success(), "`scarb lint` failed");
+
+        // TODO(#3148): Once `scarb lint` can change warning to error, we should check status instead of checking if stdout is not empty
+        // ATM `scarb lint` returns 0 even if there are warnings
+        assert!(
+            lint_output.stdout.is_empty(),
+            "`scarb lint` output should be empty"
+        );
+    }
+}
+
+#[test]
+fn validate_forge_templates() {
+    let temp_dir = assert_fs::TempDir::new().expect("Unable to create a temporary directory");
+
+    let templates = Template::value_variants();
+
+    for template in templates {
+        let package_name = format!("{}_test", template.to_string().replace('-', "_"));
+
+        runner(&temp_dir)
+            .args([
+                "new",
+                "--template",
+                template.to_string().as_str(),
+                &package_name,
+            ])
+            .assert()
+            .success();
+
+        let package_path = temp_dir.child(package_name);
+        let package_path = Utf8PathBuf::from_path_buf(package_path.to_path_buf())
+            .expect("Failed to convert to Utf8PathBuf");
+
+        // Overwrite Scarb.toml with `allow-warnings = true`
+        let scarb_toml_path = package_path.join("Scarb.toml");
+        let mut scarb_toml = fs::read_to_string(&scarb_toml_path).unwrap();
+        scarb_toml.push_str("\n[cairo]\nallow-warnings = true\n");
```

Comment:
Why do we allow them?

Reply:
We do this only to ensure that scarb check will fail if there are any warnings (that's even suggested in issue description + we've talked about it offline yesterday 😅 ) . Note that we do it only for test packages, we don't include it in `Scarb.toml` template.

Reply:
But shouldn't it be `allow_warnings=false` then?

Reply:
Oops, should be, it was a typo.